src/app/store/reducers/error.reducers.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { Action } from '@ngrx/store';
+
+import { SetErrorMessage, ClearErrorMessage } from '../actions/Error.actions';
+import { ErrorReducer, ErrorState } from './error.reducers';
+
+describe('ErrorReducer', () => {
+  let state: ErrorState;
+  let resultState: ErrorState;
+  let action: any;
+
+  describe('when initialised', () => {
+    beforeEach(() => {
+      action = {} as Action;
+      const initState: ErrorState = { errors: [] };
```

Comment:
we might also use initialErrorState for this initialErrorState.
Also, would want to create an InitialState Object that we could reference in the unit tests files so we don't hardcode the value in the test and this will act as the single source of truth?
Or have an object with a default state that matches the IAppState shape?

I think we already have an app.state.ts, could use that instead?
```
{
  router: ['/'],
  loader: false,
  errors: null,
  vehicleTechRecordModel: [],
  vehicleTestRecordModel: []
}

// in the test
import {initialState} from 'default-state'

// use the state slices to set the tests and reducers
initialState.errors
...

```

Reply:
I agree. We shoud replace line 14 with `const initState = initialAppState.error;`

Reply:
thanks, ok so we are saying we will localise defaultState and state interfaces in the reducer I believe. it is also a great approach if we wish to export them as separate module (like the redux ducks pattern) they will also have to be mapped/imported in IAppState and relevant unit test files for consistency and maintainability.
At the moment IAppState has the following if I am not wrong

```
 error?: [string] | null; // i guess this should be replaced by ErrorState as well?
```